ROX-15135: Reorganize requirements structure and add recommended requirements (#61935)
openshift/openshift-docs: modules/recommended-requirements-secured-cluster-services.adoc

// Module included in the following assemblies:
//

> // * installing/acs-recommended-requirements.adoc

Reviewer: // * cloud_service/acscs-recommended-requirements.adoc

:_content-type: CONCEPT
[id="recommended-requirements-secured-cluster-services_{context}"]
= Secured cluster services

Secured cluster services contain the following components:

* Sensor
* Admission controller
* Collector

[NOTE]
====

> Collector component is not included on this page. Required resource requirements are listed on the xref:../installing/acs-default-requirements.adoc#default-requirements-secured-cluster-services-collector_acs-default-requirements[default resource requirements page].

Reviewer: ```suggestion
Information for the Collector component is not included on this page. Required resource requirements are listed on the default resource requirements page.
```

Reviewer: We cannot have cross-references in modules, only in assembly files. We can get around this by adding an "Additional resources" section that is tied to the module and that is included in the assembly right after the module is called. The guidelines are here: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#links-hyperlinks-and-cross-references

Author: I wasn't happy about this reference because it should point to different pages depending on where a module is used. If it's used in cloud_service, then default resources for cloud_service should be linked. If it's in installing, then related default resources in installing tree should be used.

I have changed this a bit. I have removed links from modules and added "Additional resources" in the assembly doc. And in a module, I'm referring to the name used in "Additional resources".

====

[id="recommended-requirements-secured-cluster-services-sensor_{context}"]
== Sensor

Sensor monitors your Kubernetes and OpenShift Container Platform clusters. These services currently deploy in a single deployment, which handles interactions with the Kubernetes API and coordinates with Collector.

[discrete]
== Memory and CPU requirements

> The following table lists the minimum memory and CPU values required to run the Sensor on a secured cluster.

Reviewer: ```suggestion
The following table lists the minimum memory and CPU values required to run Sensor on a secured cluster.
```

Reviewer: ^Normally I see Sensor, Scanner, etc. referred to without an article, I think?

Author: It's mixed. I did a quick search. There are 64 occurrences with `the` and ~180 without.

|===
| Deployments | Pods per deployment | CPU | Memory

> | < 25k

Reviewer: ```suggestion
| < 25,000
```

Reviewer: https://www.ibm.com/docs/en/ibm-style?topic=measurement-numbers

| 3
| 2 cores
| 8 GiB

> | < 50k

Reviewer: ```suggestion
| < 50,000
```

| 3
| 2 cores
| 16 GiB
|===

[id="recommended-requirements-secured-cluster-services-admission-controller_{context}"]
== Admission controller

> The Admission controller prevents users from creating workloads that violate policies you configure.

Reviewer: ```suggestion
The Admission controller prevents users from creating workloads that violate policies that you configure.
```

[discrete]
== Memory and CPU requirements

The following table lists the minimum memory and CPU values required to run the Admission controller on a secured cluster.

|===
| Deployments | Pods per deployment | CPU | Memory

> | < 25k

Reviewer: ```suggestion
| < 25,000
```

| 3
| 0.5 cores
| 600 MiB

> | < 50k

Reviewer: ```suggestion
| < 50,000
```

// Module included in the following assemblies:
//
// * installing/acs-recommended-requirements.adoc
// * cloud_service/acscs-recommended-requirements.adoc
:_content-type: CONCEPT
[id="recommended-requirements-secured-cluster-services_{context}"]
= Secured cluster services

Secured cluster services contain the following components:

* Sensor
* Admission controller
* Collector

[NOTE]
====

> Collector component is not included on this page. Required resource requirements are listed on the *default resource requirements* page.

Reviewer: ```suggestion
Collector component is not included on this page. Required resource requirements are listed on the default resource requirements page.
```

Author: I have moved this note under the `Collector` bullet point.

====

[id="recommended-requirements-secured-cluster-services-sensor_{context}"]
== Sensor

Sensor monitors your Kubernetes and OpenShift Container Platform clusters. These services currently deploy in a single deployment, which handles interactions with the Kubernetes API and coordinates with Collector.

[discrete]
== Memory and CPU requirements

The following table lists the minimum memory and CPU values required to run Sensor on a secured cluster.

|===
| Deployments | Pods per deployment | CPU | Memory

| < 25,000
| 3
| 2 cores
| 8 GiB

| < 50,000
| 3
| 2 cores
| 16 GiB
|===

[id="recommended-requirements-secured-cluster-services-admission-controller_{context}"]
== Admission controller

> The Admission controller prevents users from creating workloads that violate policies that you configure.

Reviewer: ```suggestion
The admission controller prevents users from creating workloads that violate policies that you configure.
```

[discrete]
== Memory and CPU requirements

> The following table lists the minimum memory and CPU values required to run the Admission controller on a secured cluster.

Reviewer: ```suggestion
The following table lists the minimum memory and CPU values required to run the admission controller on a secured cluster.
```
